lte/cloud/go/services/lte/obsidian/models/conversion.go

```diff
@@ -626,3 +627,191 @@ func (m *ApnResource) getAssocs() []storage.TypeAndKey {
+		Key:          string(m.GatewayPoolID),
+		Type:         lte.GatewayPoolEntityType,
+		Config:       m.Config,
+		Name:         m.GatewayPoolName,
+		Associations: assocs,
+	}
+}
+
+func (m *GatewayPool) ToEntityUpdateCriteria() configurator.EntityUpdateCriteria {
+	return configurator.EntityUpdateCriteria{
+		Type:              lte.GatewayPoolEntityType,
+		Key:               string(m.GatewayPoolID),
+		NewName:           &m.GatewayPoolName,
+		NewConfig:         m.Config,
+		AssociationsToSet: m.getAssocs(),
+	}
+}
+
+func (m *GatewayPool) getAssocs() []storage.TypeAndKey {
+	assocs := []storage.TypeAndKey{}
+	for _, gwID := range m.GatewayIds {
+		gateway := storage.TypeAndKey{
+			Type: lte.CellularGatewayEntityType,
+			Key:  string(gwID),
+		}
+		assocs = append(assocs, gateway)
+	}
+	return assocs
+}
+
+func (m *MutableGatewayPool) ToEntityUpdateCriteria(networkID, poolID string) (configurator.EntityUpdateCriteria, error) {
+	poolEnt, err := configurator.LoadEntity(networkID, lte.GatewayPoolEntityType, poolID,
+		configurator.EntityLoadCriteria{LoadConfig: true, LoadAssocsFromThis: true}, EntitySerdes)
+	if err != nil {
+		return configurator.EntityUpdateCriteria{}, err
+	}
+	return configurator.EntityUpdateCriteria{
+		Type:              lte.GatewayPoolEntityType,
+		NewName:           &m.GatewayPoolName,
+		Key:               poolID,
+		NewConfig:         m.Config,
+		AssociationsToSet: poolEnt.Associations,
+	}, nil
+}
+
+func (m *MutableGatewayPool) ToEntity() configurator.NetworkEntity {
+	return configurator.NetworkEntity{
+		Key:    string(m.GatewayPoolID),
+		Type:   lte.GatewayPoolEntityType,
+		Config: m.Config,
+		Name:   m.GatewayPoolName,
+	}
+}
+
+func (m *GatewayPoolRecords) FromBackendModels(networkID string, gatewayID string) error {
+	cellularConfig := &GatewayCellularConfigs{}
+	err := cellularConfig.FromBackendModels(networkID, gatewayID)
+	if err != nil {
+		return err
+	}
+	*m = cellularConfig.Pooling
+	return nil
+}
+
+func (m *GatewayPoolRecords) ToUpdateCriteria(networkID, gatewayID string) ([]configurator.EntityUpdateCriteria, error) {
+	updates := []configurator.EntityUpdateCriteria{}
+	gatewayEnt, err := configurator.LoadEntity(networkID, lte.CellularGatewayEntityType, gatewayID,
+		configurator.EntityLoadCriteria{LoadAssocsToThis: true},
+		EntitySerdes,
+	)
+	if err != nil {
+		return nil, err
+	}
+	oldPoolIds := []GatewayPoolID{}
+	// Get all existing parent associations that are of type gateway pool
+	for _, parentAssoc := range gatewayEnt.ParentAssociations {
+		if parentAssoc.Type == lte.GatewayPoolEntityType {
+			oldPoolIds = append(oldPoolIds, GatewayPoolID(parentAssoc.Key))
+		}
+	}
+	newPoolIds := []GatewayPoolID{}
+	for _, record := range *m {
+		newPoolIds = append(newPoolIds, record.GatewayPoolID)
+	}
+	err = validateNewGatewayPools(networkID, newPoolIds)
+	if err != nil {
+		return nil, err
+	}
+
+	idsToDelete := getGatewayPoolSetDifference(oldPoolIds, newPoolIds)
```

Comment:
Don't use a bespoke fn here -- there's already `aLessB, bLessA := funk.DifferenceString(a, b)`. You might have to do some type coercion (or just use `funk.Difference`) but this way is cleaner

See e.g. services/lte/obsidian/models/conversion.go:290